Add more logging to Tools/Scripts/delete-if-too-large

chgibb-apple opened the pull request:
#### 738028099ce467df421c045d54dada3d9d1b9ff2
<pre>
Add more logging to Tools/Scripts/delete-if-too-large
<a href="https://bugs.webkit.org/show_bug.cgi?id=245489">https://bugs.webkit.org/show_bug.cgi?id=245489</a>
rdar://99293146

Reviewed by Aakash Jain.

* Tools/Scripts/delete-if-too-large:

Canonical link: <a href="https://commits.webkit.org/255737@main">https://commits.webkit.org/255737@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1fa7e4c5d61600f0db33600dfa9820b9dff44053

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/91496 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/617 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/22141 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/101226 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/161260 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/95501 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/649 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/29386 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83842 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/97578 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/97154 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/415 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/78210 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/27375 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/82339 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/81995 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/70413 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/35626 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/16024 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/33410 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/17117 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/37215 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/41183 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1857 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39131 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/36234 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -216,6 +216,15 @@ class PruneCoreSymbolicationdCacheIfTooLarge(shell.ShellCommand):
                "/System/Library/Caches/com.apple.coresymbolicationd"]
 
 
+class PruneCoreSymbolicationdDataIfTooLarge(shell.ShellCommand):
```
This PR is just adding the step. Need to update factory as well to invoke this step.

chgibb-apple replied:
Indeed. I opened the PR early to get feedback from the Python linter. Related, I implemented the most naive solution to this problem. I'm not sure how to best go about parameterising `PruneCoreSymbolicationdCacheIfTooLarge` within our use of buildbot.

aj062 commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -216,6 +216,15 @@ class PruneCoreSymbolicationdCacheIfTooLarge(shell.ShellCommand):
+    flunkOnFailure = False
+    haltOnFailure = False
+    command = ["python3", "Tools/Scripts/delete-if-too-large",
+               "/System/Volumes/Data/System/Library/Caches/com.apple.coresymbolicationd"]
```
Running the same step/script twice just to pass two different arguments doesn't seems right. Either the script should accept multiple parameters and handle them appropriately, or the paths can be hard-coded inside the script itself.

chgibb-apple replied:
I agree. 

> Related, I implemented the most naive solution to this problem. I'm not sure how to best go about parameterising PruneCoreSymbolicationdCacheIfTooLarge within our use of buildbot.

How can I go about parameterising PruneCoreSymbolicationdCacheIfTooLarge? It's not clear to me if we can actually  (or should be )pass arguments or if the way we compose out list of steps is meant to be purely declarative.

aj062 replied:
> How can I go about parameterising PruneCoreSymbolicationdCacheIfTooLarge? 

In steps.py for the command for PruneCoreSymbolicationdCacheIfTooLarge You can pass more arguments, just like you are passing one argument currently. 

You can modify delete-if-too-large script so that users can pass multiple directories. You can either use argument parser (like https://docs.python.org/dev/library/argparse.html) or for this simple user-case, you can simply loop over sys.argv and store the arguments in a list of directories. Then just loop over that list and execute your logic to delete.

If the list of directories is going to be fixed, other option is to hard-code these directories names inside the script.

aproskuryakov commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -216,6 +216,15 @@ class PruneCoreSymbolicationdCacheIfTooLarge(shell.ShellCommand):
+    flunkOnFailure = False
+    haltOnFailure = False
+    command = ["python3", "Tools/Scripts/delete-if-too-large",
+               "/System/Volumes/Data/System/Library/Caches/com.apple.coresymbolicationd"]
```
Isn't `/System/Volumes/Data/System/Library/Caches` just a worse way to write `/System/Library/Caches`?

chgibb-apple replied:
I believe it is. Though, for some reason we are seeing both of these paths needed.

aproskuryakov replied:
I think that's the same directory, just linked to two places in the hierarchy in some way.

chgibb-apple replied:
There's some machines in the fleet that seem to need this path for some reason.

aj062 commented on Tools/Scripts/delete-if-too-large:
```diff
@@ -29,21 +29,24 @@ from pathlib import Path
+            os.remove(directory)
+            print("Deleted " + directory)
+        else:
+            print(f'{directory} is {directorySize}. Doing nothing') 
```
Nit: please add unit (kb/mb/gb) in this message.

aj062 replied:
this comment about adding unit is still applicable.

chgibb-apple committed:
Add more logging to Tools/Scripts/delete-if-too-large
https://bugs.webkit.org/show_bug.cgi?id=245489
rdar://99293146

Reviewed by Aakash Jain.

* Tools/Scripts/delete-if-too-large:

Canonical link: https://commits.webkit.org/255737@main
